main.tf

```diff
@@ -131,6 +131,14 @@ module "gcr_project_roles" {
   sa_depends_on = module.services_sa.email
 }
 
+module "custom_external_roles" {
+  source = "./modules/external-roles"
+
+  roles_map     = var.custom_external_roles
+  project_id    = module.project_factory.project_id
+  sa_depends_on = module.services_sa.email
```

Comment:
why does it depend on services_sa.email? can't you add these custom roles to any specified SA?